File: packages/ts/src/code-toolbox/WorkspacePythonCodeToolbox.ts
import { WorkspaceCodeToolbox } from "../Workspace";

export class WorkspacePythonCodeToolbox implements WorkspaceCodeToolbox {
    public getDefaultImage(): string {
        return 'ai-test:1'    //  todo: replace with python image
    }

    public getRunCommand(code: string): string {
        const base64Code = Buffer.from(code).toString('base64');
        return `python3 -c "exec(__import__('base64').b64decode('${base64Code}').decode())"`;
    }
}

File: packages/python/src/daytona_sdk/code_toolbox/workspace_python_code_toolbox.py
import base64
from typing import Protocol


class WorkspaceCodeToolbox(Protocol):
    def get_default_image(self) -> str: ...
    def get_run_command(self, code: str) -> str: ...


class WorkspacePythonCodeToolbox:
    def get_default_image(self) -> str:
        return "ai-test:1"  # todo: replace with python image

    def get_run_command(self, code: str) -> str:
        base64_code = base64.b64encode(code.encode()).decode()
        return f"python3 -c \"exec(__import__('base64').b64decode('{base64_code}').decode())\""
